src/userdev/java/net/minecraftforge/gradle/userdev/MinecraftUserRepo.java

```diff
@@ -501,6 +531,18 @@ private File findRaw(String mapping) throws IOException {
                 while ((entry = zin.getNextEntry()) != null) {
                     if (!entry.getName().startsWith(prefix) || entry.isDirectory())
                         continue;
+
+                    // If an entry has a specific side in its name, don't apply
+                    // it when we're on the opposite side. Entries without a specific
+                    // side should always be applied
+                    if ("server".equals(NAME) && entry.getName().contains("/client/")) {
```

Comment:
Patchers are only joined. So this filter is unneeded. The InjectFunction is run on the MC code itself so that filter works fine on this repo as well.

Reply:
This filter is definitely necessary - I get a compilation error when using `UserDevPlugin` otherwise. The zip entries come directly from the `mcp` jar, which contains a [Start](https://github.com/MinecraftForge/MCPConfig/blob/2d1b27131651897791bdee0f1574f76467f7ebb0/versions/1.13/inject/mcp/client/Start.java) class that we don't want to inject into a server jar.

Reply:
Actually got time to verify this, and I was wrong I thought you were in the other loop where we inject the patcher's code.
This is just for the MCP jar so the filter is fine.